crates/goose-cli/src/main.rs

```diff
@@ -62,45 +62,136 @@ struct Cli {
 
 #[derive(Subcommand)]
 enum Command {
+    /// Configure Goose settings and profiles
+    #[command(about = "Configure Goose settings and profiles")]
     Configure {
+        /// Name of the profile to configure
+        #[arg(
+            help = "Profile name to configure",
+            long_help = "Create or modify a named configuration profile. Use 'default' for the default profile."
+        )]
         profile_name: Option<String>,
     },
+
+    /// Manage system prompts and behaviors
+    #[command(about = "Manage system prompts and behaviors")]
     System {
         #[command(subcommand)]
         action: SystemCommands,
     },
-    /// Start or resume sessions with an optional session name
+
+    /// Start or resume interactive chat sessions
+    #[command(
+        about = "Start or resume interactive chat sessions",
+        alias = "s",
+    )]
     Session {
-        #[arg(short, long)]
+        /// Name for the chat session
+        #[arg(
+            short,
+            long,
+            value_name = "NAME",
+            help = "Name for the chat session (e.g., 'project-x')",
+            long_help = "Specify a name for your chat session. When used with --resume, will resume this specific session if it exists."
+        )]
         session: Option<String>,
-        #[arg(short, long)]
+
+        /// Configuration profile to use
+        #[arg(
+            short,
+            long,
+            value_name = "PROFILE",
+            help = "Configuration profile to use (e.g., 'default')",
+            long_help = "Use a specific configuration profile. Profiles contain settings like API keys and model preferences."
+        )]
         profile: Option<String>,
-        #[arg(short, long, action = clap::ArgAction::SetTrue)]
+
+        /// Resume a previous session
+        #[arg(
+            short,
+            long,
+            help = "Resume a previous session (last used or specified by --session)",
+            long_help = "Continue from a previous chat session. If --session is provided, resumes that specific session. Otherwise resumes the last used session."
```

Comment:
cc: @jsibbison-square to confirm the various combinations work as described here?